Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -732,20 +740,40 @@ const AXTextRuns* AXTextMarker::runs() const
     return object ? object->textRuns() : nullptr;
 }
 
-AXTextMarker AXTextMarker::findMarker(AXDirection direction, std::optional<AXID> stopAtID) const
+AXTextMarker AXTextMarker::findMarker(AXDirection direction, bool coalesceObjectBreaks, bool ignoreBRs, std::optional<AXID> stopAtID) const
 {
+    // This method has two boolean options:
+    // - coalesceObjectBreaks: Mimics behavior from textMarkerDataForNextCharacterOffset, where we skip nodes
+    //   that have the same visual position (i.e., there is 0 length between them). When false, we traverse all
+    //   possible text markers (which is important for searching)
+    // - ignoreBRs: In most cases, we want to skip <br> tags when not in an editable context. This is not true,
+    //   for example, when computing text marker indexes.
+
     if (!isValid())
         return { };
     if (!isInTextRun())
-        return toTextRunMarker().findMarker(direction, stopAtID);
+        return toTextRunMarker().findMarker(direction, coalesceObjectBreaks, ignoreBRs, stopAtID);
 
-    if ((direction == AXDirection::Next && offset() < runs()->totalLength()) || (direction == AXDirection::Previous && offset() > 0))
-        return { treeID(), objectID(), direction == AXDirection::Next ? offset() + 1 : offset() - 1 };
+    RefPtr object = isolatedObject();
+
+    // If the BR isn't in an editable ancestor, we shouldn't be including it (in most cases of findMarker).
+    bool shouldSkipBR = (ignoreBRs && !object->editableAncestor()) && object && object->roleValue() == AccessibilityRole::LineBreak;
```

Comment:
`object && object->roleValue() == AccessibilityRole::LineBreak` is significantly cheaper of a check vs. `editableAncestor()` which does an ancestry traversal, so let's put that part first. This ensures we only do the expensive work when we really have to (for line breaks).